sdk/timeseriesinsights/Azure.IoT.TimeSeriesInsights/src/Models/TimeSeriesValue.cs

```diff
@@ -0,0 +1,528 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Azure.IoT.TimeSeriesInsights
+{
+    /// <summary>
+    /// A variant type that represents a Time Series value that is retrieved when executing Query requests.
+    /// </summary>
+    public readonly struct TimeSeriesValue
+    {
+        private static readonly object s_nullInt32 = new();
+        private static readonly object s_nullDouble = new();
+        private static readonly object s_nullBoolean = new();
+        private static readonly object s_nullDateTimeOffset = new();
+        private static readonly object s_nullTimeSpan = new();
+
+        private readonly long _i64;
+        private readonly object _obj;
+
+        /// <summary>
+        /// Get the type of the variant.
+        /// </summary>
+        public Type Type
+        {
+            get
+            {
+                if (_obj.GetType() == typeof(string))
+                    return typeof(string);
+                if (_obj.GetType() == typeof(byte[]))
+                    return typeof(string);
+                if (_obj.GetType() == typeof(DateTimeOffset))
+                    return typeof(DateTimeOffset);
+                if (_obj.GetType() == typeof(object))
+                {
+                    if (_obj == s_nullInt32)
+                        return typeof(int?);
+                    if (_obj == s_nullDouble)
+                        return typeof(double?);
+                    if (_obj == s_nullBoolean)
+                        return typeof(bool?);
+                    if (_obj == s_nullDateTimeOffset)
+                        return typeof(DateTimeOffset?);
+                }
+                return (Type)_obj;
+            }
+        }
+
+        #region Int32
```

Comment:
Is this used more for sections and making code readable?

Reply:
Yes